lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -114,16 +115,77 @@ def define_let_it_be_methods(identifier, **modifiers)
 
       let(identifier, &let_accessor)
     end
+
+    module Freezer
+      class << self
+        # Rerucsively freezes the object to detect modifications
+        def deep_freeze(record)
+          return if record.frozen?
+          return if Stoplist.include?(record)
+
+          record.freeze
+
+          # Support `let_it_be` with `create_list`
+          return record.each { |rec| deep_freeze(rec) } if record.respond_to?(:each)
+
+          # Freeze associations as well.
+          # NOTE: `reload` statements in test or production code will cause
+          # a `FrozenError`. In case the use of `reload` cannot be avoided, use
+          # `reload: true` in `let_it_be` declaration.
+          return unless defined?(::ActiveRecord::Base)
+          return unless record.is_a?(::ActiveRecord::Base)
+
+          record.class.reflections.keys.each do |reflection|
+            # But only if they are already loaded. If not yet loaded, they weren't
+            # created by factories, and it's ok to mutate them.
+            next unless record.association(reflection.to_sym).loaded?
+
+            target = record.association(reflection.to_sym).target
+            deep_freeze(target) if target.is_a?(::ActiveRecord::Base) || target.respond_to?(:each)
+          end
+        end
+      end
+    end
+
+    # Stoplist to prevent freezing objects that are defined with `let_it_be`'s
+    # `reload: true`/`refind: true`/`freeze: false` options during deep freezing.
+    # To only keep track of objects that are available in current example group,
+    # `begin` adds a new layer, and `rollback` removes a layer of unrelated objects
+    # along with rolling back the transaction where they were created.
+    module Stoplist
+      class << self
+        def include?(record)
+          @stoplist.any? { |layer| layer.include?(record) }
+        end
+
+        def push(record)
+          @stoplist.last.push(record)
+        end
+
+        def begin
+          @stoplist.push([])
+        end
+
+        def rollback
+          @stoplist.pop
+        end
+      end
+
+      @stoplist = [] # Stack of example group-related variable definitions
+    end
   end
 end
 
-if defined?(::ActiveRecord)
+if defined?(::ActiveRecord::Base)
```

Comment:
As you've mentioned above, having `ActiveRecord` in place is not a sign that it's used.